packages/core/src/actions/smartAccount/buildUserOperation.ts

```diff
@@ -60,15 +53,23 @@ export async function buildUserOperation<
   const _uo =
     entryPoint.version === "0.6.0"
       ? ({
-          initCode: getInitCode,
+          initCode: account.getInitCode(),
           sender: account.address,
           nonce,
           callData,
           signature,
         } as Deferrable<UserOperationStruct<TEntryPointVersion>>)
       : ({
-          factory: getFactoryAndData.then(([factory]) => factory),
-          factoryData: getFactoryAndData.then(([, data]) => data),
+          factory: account
+            .isAccountDeployed()
+            .then((deployed) =>
+              deployed ? account.getFactoryAddress() : undefined
```

Comment:
Are these backwards? I think you want the factory address if you're _not_ deployed.

Reply:
same with data below it seems